### ISSUE TITLE
gcc prints buffer uninitialized warnings from png library.

### ISSUE THREAD

**Issue by rageworx** [fltk/fltk#757]
**Describe the bug**
- I found gcc compilation warnings occurs from png library ( fltk/png )
- These are just warning, but I believe it should be fixed by little effort.

**To Reproduce**
- Configure FLTK to use fltk-png.
- Build on gcc 13.1.0 (rev7, MinGW-W64)
- g++ will be prints warnings while compiling PNG.

**Expected behavior**
I can solve these warnings to insert array initializer `={0};`.
Change log is here : https://github.com/rageworx/fltk-custom/commit/9f17e1fb601595885e5494cbd7460b850f147405

**Log**
```
In file included from pngpriv.h:77,
                 from pngerror.c:19:
pngerror.c: In function 'fltk_png_warning_parameter_unsigned':
pngprefix.h:147:27: warning: 'buffer' may be used uninitialized [-Wmaybe-uniniti
alized]
  147 | #define png_format_number fltk_png_format_number
pngpriv.h:1749:4: note: in expansion of macro 'png_format_number'
 1749 |    png_format_number(buffer, buffer + (sizeof buffer), format, number)
      |    ^~~~~~~~~~~~~~~~~
pngerror.c:259:37: note: in expansion of macro 'PNG_FORMAT_NUMBER'
  259 |    png_warning_parameter(p, number, PNG_FORMAT_NUMBER(buffer, format, va
lue));
      |                                     ^~~~~~~~~~~~~~~~~
pngprefix.h:147:27: note: by argument 1 of type 'png_const_charp' {aka 'const ch
ar *'} to 'fltk_png_format_number' declared here
  147 | #define png_format_number fltk_png_format_number
      |                           ^~~~~~~~~~~~~~~~~~~~~~
pngerror.c:133:1: note: in expansion of macro 'png_format_number'
  133 | png_format_number(png_const_charp start, png_charp end, int format,
      | ^~~~~~~~~~~~~~~~~
pngerror.c:258:9: note: 'buffer' declared here
  258 |    char buffer[PNG_NUMBER_BUFFER_SIZE];
      |         ^~~~~~
pngerror.c: In function 'fltk_png_warning_parameter_signed':
pngprefix.h:147:27: warning: 'buffer' may be used uninitialized [-Wmaybe-uniniti
alized]
  147 | #define png_format_number fltk_png_format_number
pngpriv.h:1749:4: note: in expansion of macro 'png_format_number'
 1749 |    png_format_number(buffer, buffer + (sizeof buffer), format, number)
      |    ^~~~~~~~~~~~~~~~~
pngerror.c:275:10: note: in expansion of macro 'PNG_FORMAT_NUMBER'
  275 |    str = PNG_FORMAT_NUMBER(buffer, format, u);
      |          ^~~~~~~~~~~~~~~~~
pngprefix.h:147:27: note: by argument 1 of type 'png_const_charp' {aka 'const ch
ar *'} to 'fltk_png_format_number' declared here
  147 | #define png_format_number fltk_png_format_number
      |                           ^~~~~~~~~~~~~~~~~~~~~~
pngerror.c:133:1: note: in expansion of macro 'png_format_number'
  133 | png_format_number(png_const_charp start, png_charp end, int format,
      | ^~~~~~~~~~~~~~~~~
pngerror.c:268:9: note: 'buffer' declared here
  268 |    char buffer[PNG_NUMBER_BUFFER_SIZE];
      |         ^~~~~~
```

**FLTK Version**
Please complete the following information and delete non-applicable lines:
 - Version: [e.g. 1.4.0]
 - If from Git, branch: master

**FLTK Configure / Build Options**
```
EFLAGS="--enable-localjpeg --enable-localzlib --enable-localpng"
BSYS="--build=x86_64-w64-mingw32"
./configure ${BSYS} ${EFLAGS} --enable-debug
```

**Operating System / Platform:**
 - OS: Windows 10
 - OS Version: Microsoft Windows [Version 10.0.19042.1706]
 - Processor: Intel i7-7700

**Comment by rageworx** [comment]
I made a PR for this, https://github.com/fltk/fltk/pull/758.

**Comment by rageworx** [comment]
Ok ! I got it.

**Comment by rageworx** [comment]
I closed this issue for now, and hope will open again if you need ;)

**Comment by Albrecht-S** [comment]
Fixed in commit 1e890ea4e4ba00a66743cc97b89dfd4991167ce1.
